.gitignore

```diff
@@ -9,6 +9,7 @@
 /TAGS
 .ghc.environment*
 /cabal.project.local
+**stack.yaml.lock
```

Comment:
stack.yaml.lock helps to reproduce builds when packages on Hackage have small patches without version change. There is no need to ignore it.

Reply:
`stack.yaml.lock` wasn't under version control, and I was tired of avoiding `git add .`.  I am more than happy to drop the commit in a rebase before merging, but if the file is important I would prefer to include it in the version control.

Reply:
OK, leave it ignored

Reply:
I know too little about Stack nowadays, but for lockfiles in general, they're useful for applications rather than libraries. If we can't disable it in the Stack configuration, I think it should be at least ignored.

Reply:
I looked for some ways to avoid the lock file with stack, but I couldn't find anything that would do it... :/

So I guess ignoring it will be the way forward.